core/bin/core_api.rs

```diff
@@ -3076,15 +3082,23 @@ async fn folders_delete(
                         Some(e),
                     )
                 }
-                Ok(_) => (
-                    StatusCode::OK,
-                    Json(APIResponse {
-                        error: None,
-                        response: Some(json!({
-                            "success": true,
-                        })),
-                    }),
-                ),
+                Ok(_) => match state.search_store.delete_node(folder_id).await {
```

Comment:
Smells like we want a Folder object that handles the delete logic :)

Reply:
to be consistent with the absence of consitency (since document does not do delete logic via an object :) ) I made the simplest most straightforward code :)
but happy to change

Reply:
The succession of api error handling is a bit distasteful here. Won't block on it of course

Reply:
true, refactored to avoid that